soh/src/overlays/actors/ovl_En_Tk/z_en_tk.c

```diff
@@ -502,8 +502,12 @@ void EnTk_Init(Actor* thisx, PlayState* play) {
 
     CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
 
-    if (gSaveContext.dayTime <= 0xC000 || gSaveContext.dayTime >= 0xE000 || !!LINK_IS_ADULT ||
-        play->sceneNum != SCENE_SPOT02) {
+    if (CVarGetInteger("gDampeAllNight", 0)) {
+        if (!!LINK_IS_ADULT || play->sceneNum != SCENE_SPOT02) {
```

Comment:
is there a reason you're using `!!` here? `LINK_IS_ADULT` is
```c
#define LINK_IS_ADULT (gSaveContext.linkAge == LINK_AGE_ADULT)
```
so it should already be getting us a value that works for a boolean check

Reply:
nevermind, i see now that you just copied it over from the authentic logic, should be fine